contracts/ClaimsManager.sol

```diff
@@ -1,5 +1,111 @@
 //SPDX-License-Identifier: MIT
-pragma solidity ^0.8.0;
+
+pragma solidity >=0.8.4;
+
+/// @title API3 Service Coverage Claims Manager
+/// @notice contract manager of API3 service coverage claims process and payments, permitted to withdraw as many tokens as necessary from API3's staking pool to satisfy successful valid claims on service coverage
+/// @dev the primary DAO Agent must call setClaimsManagerStatus(coverageClaimManager, true) so this contract will satisfy the onlyClaimsManager() modifier to pay out claims or counteroffers
+
+interface IAPI3Pool {
+    function payOutClaim(address recipient, uint256 amount) external;
+}
 
 contract ClaimsManager {
+
+    address constant Api3Pool = 0x6dd655f10d4b9E242aE186D9050B68F725c76d76; // API3 DAO staking pool which collateralizes the service coverage
+    address immutable coverageClaimManager;
+    uint256 claimID;
+    IAPI3Pool iAPI3Pool;
+    mapping(address => bool) isCounterofferor; // counterofferor(s) of service coverage claims, i.e. multisig agent of API3
+    mapping(address => bool) isResolver; // resolver(s) of service coverage claims, i.e. Kleros
+    mapping(uint256 => CoverageClaim) public claims;
+
+    enum CoverageClaimStatus { None, Submitted, CounterOffered, Resolved }
+
+    struct CoverageClaim {
+        uint256 claimedDamagesAmount; // amount of damages claimed by claimant in API3 tokens as of the time of claim submission
+        uint256 counterofferAmount; // counteroffer amount of API3 tokens by counterofferor
+        uint256 timestamp; // claim submission timestamp
+        address claimant; // holder of a service coverage policy making the applicable claim
+        address beneficiary; // absent cryptographic proof of privity with claimant, should be same address as claimant, and currently hardcoded accordingly in submitClaim()
```

Comment:
People who read code are usually ESL, this kind of language is not suitable for documentation purposes

Reply:
so should I just remove the comments entirely? they were mostly explanatory for the initial commit anyway

Reply:
I prefer not having any comments during development because they will keep getting out of date. When we add them back they should be grug-tier.